pkg/hns/endpoint_windows.go

```
@@ -40,6 +51,92 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) *hcsshim.HNSEndpoint {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
+	if hnsEndpoint != nil && err != nil {
+		if hnsEndpoint.VirtualNetwork != epInfo.NetworkId {
+			_, err = hnsEndpoint.Delete()
+			if err != nil {
+				errors.Annotatef(err, "[win-cni] Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
+			}
+		} else {
+			errors.Annotatef(err, "[win-cni] Found existing endpoint %v", epInfo.EndpointName)
+		}
+	} else {
+
+		hnsEndpoint = &hcsshim.HNSEndpoint{
+			Name:           epInfo.EndpointName,
+			VirtualNetwork: epInfo.NetworkId,
+			DNSServerList:  strings.Join(epInfo.Nameservers, ","),
+			DNSSuffix:      strings.Join(epInfo.DnsSearch, ","),
+			GatewayAddress: epInfo.Gateway.String(),
+			IPAddress:      epInfo.IpAddress,
+			Policies:       n.MarshalPolicies(),
+		}
+
+		errors.Annotatef(err, "Adding Hns Endpoint %v", hnsEndpoint)
```

This error is not being handled